### PR TITLE
[wip] Ng bulk adding

### PR DESCRIPTION
**Pull request by ghost**
working version

Connected to #1606

### CODE REVIEW

**Comment by damian1976** [comment]
IMHO when bulk adding assets I would change the old method relying on 2 textareas(SN + Barcode)  into triples of 'SN' + 'Barcode' + 'Position' next to each other. In order to add next triple '+' Button would exist underneath. By clicking on we would activate the next triple below. In this way, in one step we can create new assets and place them correctly inside the rack. Additionally when clicking on '+', 'Position' 
field in **Location Info** would have to be ignored/disabled to let the user know that positions are now read from these located next to SN/Barcode textfields.
Does it sound reasonably?
Anyhow, the idea is to introduce 'Position' field to every asset added in 'Bulk add' mode.

File: src/ralph/admin/tests/tests_functional.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+# -*- cmding: utf-8 -*-
```
Comment (vi4m) [inline src/ralph/admin/tests/tests_functional.py:1]: ?

**Comment by damian1976** [comment]
If you disagree with the above could you at least add position to sn/Barcode. It would really help in bulk add....

File: src/ralph/admin/fields.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+from django.forms.models import save_instance
+from django import forms
+from django.db import transaction
+from django.forms.util import ErrorList
+from django.utils.translation import ugettext as _
+
+
+class MultilineField(forms.CharField):
+    """
+    This field is a textarea which treats its content as many values seperated
```
Comment (vi4m) [inline src/ralph/admin/fields.py:15]: Well done descriptions across entire pull request. Very good!

**Comment by vi4m** [comment]
@damian1976 we have to make sure we need this functionality too - we will ask our operations team.

**Comment by damian1976** [comment]
@vi4m Did you by chance ask them about this functionality?
maybe I'm wron but how do you bulk add e.g 20 identical servers?You ad1 asset with 20 SNs, then after creating them you search them by service/env or similar, and finally go one by one setting valid position? My proposition would make it easier and would shortened the whole process to a single step...

**Comment by ghost** [comment]
@damian1976 Your case could be solved by new concept (https://github.com/allegro/ralph/pull/1721). However nothing was settled and concept is still work in progress. Nonetheless the case you described should be resolved in comfortable way for the user. Editing tons of assets indyvidually for `position` or `slot number` (similar case IMHO) is unacceptable.

**Comment by damian1976** [comment]
@xliiv OK, thanks. I'll keep the fingers crossed for the proposed solution:) Hope, this will facilitate the bulk adding process and setting assets in correct position inside rack without editing them continuously and setting location info once again one by one. Is it possible to prepare (testing version of course) this concept for DC asset as well ASAP?

**Pull request closed without merging**